TST: sparse.linalg.lobpcg: add unit tests for accuracy

lobpcg opened the pull request:
#### Reference issue
Closes gh-19355

#### What does this implement/fix?
see gh-19355

#### Additional information
supersedes closed https://github.com/scipy/scipy/pull/19356

lobpcg committed:
add more tests test_lobpcg.py

lobpcg commented:
Windows tests / fail slow, fast, py3.12/npAny, dev.py (pull_request) Failing after 35m with
```
[gw0] win32 -- Python 3.12.4 C:\hostedtoolcache\windows\Python\3.12.4\x64\python.exe
Test passed but took too long to run: Duration 1.0849475999998504s > 1.0s
```

I am unsure what to do about it.

mdhaber commented:
Typical tests that execute in every job of every CI run take <0.25s to run (and often much less). This new test takes longer than that, so the failure asks you to consider whether the test needs to be run so often. Options include speeding up the test, marking it with `@pytest.mark.slow` (so it runs on only some jobs), or giving it an exception to the time limit like `@pytest.mark.fail_slow(10)` (if it is particularly important that it run on all platforms).

lobpcg committed:
@pytest.mark.fail_slow(10) in sakurai test_lobpcg.py

Fixing 
[gw0] win32 -- Python 3.12.4 C:\hostedtoolcache\windows\Python\3.12.4\x64\python.exe
Test passed but took too long to run: Duration 1.0849475999998504s > 1.0s
as proposed in https://github.com/scipy/scipy/pull/21220#issuecomment-2246797421

lobpcg committed:
Merge branch 'scipy:main' into patch-5

j-bowhay committed:
Merge branch 'main' into patch-5

j-bowhay committed:
TST: mark xslow
